Removing the deprecated methods and renaming the ones those have the … (#114)
immutable/imx-core-sdk: README.md

// Use the helper function to get the config
const config = getConfig(ethNetwork);

// Setup a provider and a signer
const privateKey = YOUR_PRIVATE_KEY;
const provider = new AlchemyProvider(ethNetwork, YOUR_ALCHEMY_API_KEY);
const signer = new Wallet(privateKey).connect(provider);
```

#### WalletConnection

> WalletConnection is a top level connector which contains all the needed resources (eg.: signers for L1 and L2) to possibilitate Core SDK workflows usage.

Reviewer: @sprngbk can you please help polish the wording here 🙏

Author: > "WalletConnection is a high-level connector that contains all necessary resources (eg.: signers for L1 and L2) that make it possible to operate through the Core SDK workflows."

@Kastet  Wdyt?

// Use the helper function to get the config
const config = getConfig(ethNetwork);

// Setup a provider and a signer
const privateKey = YOUR_PRIVATE_KEY;
const provider = new AlchemyProvider(ethNetwork, YOUR_ALCHEMY_API_KEY);
const signer = new Wallet(privateKey).connect(provider);
```

#### WalletConnection

> WalletConnection is a top level connector which contains all the needed resources (eg.: signers for L1 and L2) to possibilitate Core SDK workflows usage.

Reviewer 2: ```suggestion
WalletConnection is a top level connector which contains all the required resources (eg. L1 and L2 signers) to enable usage of the Core SDK workflows.
```

// User registration workflow example
import { AlchemyProvider } from '@ethersproject/providers';
import { Wallet } from '@ethersproject/wallet';
import { 
  Workflows, 
  BaseSigner, 
  getConfig, 
  generateStarkWallet,
} from '@imtbl/core-sdk';

const ethNetwork = 'ropsten';

// Sets up the provider
const alchemyApiKey = 'UPDATE WITH THE ALCHEMY API KEY HERE';
const alchemyProvider = new AlchemyProvider(ethNetwork, alchemyApiKey);

// Sets up the L1Signer
const privateKey = 'UPDATE WITH THE PRIVATE KEY HERE';
const l1Wallet = new Wallet(privateKey);
const l1Signer = l1Wallet.connect(alchemyProvider);

// Sets up the L2Signer

> const l2Wallet = await generateStarkWallet(l1Signer);

Reviewer: @RenatoChitolina is that intentionally called `l2Wallet`? Should this be just `l2Signer` to be consistent with the rest of the code?
